tests/test_sync_http_client.py

```diff
@@ -251,6 +252,23 @@ def test_request_bad_body_raises_expected_exception_with_request_data(self):
             # This'll fail for sure here but... just using the nice error that'd come up
             assert ex.__class__ == ServerException
 
+    def test_conflict_exception(self):
+        request_id = "request-123"
+
+        self.http_client._client.request = MagicMock(
+            return_value=httpx.Response(
+                status_code=409,
+                headers={"X-Request-ID": request_id},
+            ),
+        )
+
+        try:
+            self.http_client.request("bad_place")
+        except ConflictException as ex:
+            assert str(ex) == "(message=No message, request_id=request-123)"
+        except Exception as ex:
+            assert ex.__class__ == ConflictException
```

Comment:
Do you know why we do this multiple-`except` blocks in this test file?

IIRC Python will only call the first matching `except` block so this `except Exception` block will never be entered.

Reply:
That's a great question. After changing the test it appears there's no reason really. Let me update these.

Reply:
Also confirming one of these tests failing doesn't look any different.